Add TCP Metrics Task

douglas-reid opened the pull request:
This PR adds a Task to the docs on collecting TCP Metrics. It provides a small overview of TCP Attribute flow (which differs from HTTP) and exercises the MongoDB part of the BookInfo sample.

Fixes: istio/istio#766

istio-testing commented:
Jenkins job istio.github.io/presubmit passed

rshriram commented on _docs/concepts/policy-and-control/tcp-services.md:
```diff
@@ -0,0 +1,80 @@
+
+## Overview
+
+As of the 0.2 release, Istio supports adding TCP services to an Istio Service
```
Nit, we have always had support for tcp in Pilot. So, i would scope this sentence to "support for enforcing policies on plain TCP services". Or probaby even get rid of it.

rshriram replied:
See release notes for how we phrased this.

rshriram commented on _docs/concepts/policy-and-control/tcp-services.md:
```diff
@@ -0,0 +1,80 @@
+
+## TCP Attributes
+
+Several tcp-specific attributes enable TCP policy and control within Istio.
```
TCP-specific

Rachael-Graham commented on _docs/concepts/policy-and-control/tcp-services.md:
```diff
@@ -0,0 +1,80 @@
+
+The following configuration establishes basic access control for an example
+MongoDB service, based on `source.ip`. Here, we only allow TCP connections to
+the mongodb service coming from `10.1.10.1`.
```
MongoDB

kyessenov commented on _docs/concepts/policy-and-control/tcp-services.md:
```diff
@@ -0,0 +1,80 @@
+  name: mongoaccessctl
+  namespace: istio-system
+spec:
+  match: 'context.protocol == "tcp" 
```
Remove single quotes around - are they necessary?

kyessenov commented on _docs/concepts/policy-and-control/tcp-services.md:
```diff
@@ -0,0 +1,80 @@
+    source_version: source.labels["version"] | "unknown"
+    destination_service: destination.service | "unknown"
+    destination_version: destination.labels["version"] | "unknown"
+  monitored_resource_type: '"UNSPECIFIED"'
```
We strive for camel-case version throughout: `monitoredResourceType`

douglas-reid commented:
PTAL. I believe I have addressed all of the review feedback:
- changed colors in sequence diagram
- camelCase for `monitoredResourceType`
- removed problematic description of 0.2 TCP support (leave that for release notes)
- caps for TCP

istio-testing commented:
Jenkins job istio.github.io/presubmit passed
